src/lib/api.js

```diff
@@ -211,3 +216,35 @@ export const getPDFReport = (userId, email) => {
     mode: 'cors'
   });
 };
+
+// only make network request for readers when we have none
+// populateReaders (dispatch: Function, getState: Function) => Promise
+const populateReaders = (dispatch, getState) => {
+  const readersCount = getReaders(getState()).size;
+  return readersCount ? Promise.resolve : readersRequest()(dispatch, getState);
```

Comment:
Nope, you had it right the first time. I should be returning a Promise here (and I've documented it as such), but I'm accidentally returning a Function instead. The only reason this works at all is because I'm using it with `Promise.all()` further down which accepts and Array of any old thing, where truthy values in the array are automatically wrapped into resolved Promises.